### DISCUSSION TITLE
More elegant responsive styles API

### DISCUSSION THREAD

**Discussion by nk1tz** [mantinedev/mantine#3316] [Feature requests]
With other styled-system based components libraries, the following responsive styles API is offered:

```typescript
<Group noWrap={{ xs: false, sm: true }}>
// or
<Group sx={{ flexWrap: { xs: 'wrap', sm: 'nowrap' } }}>
// or 
<Group noWrap={[false, true]} >
 ```
 
 In Mantine, the most elegant way to achieve the same is to do the following:
 ```typescript
 <Group
  sx={{
    [`@media (max-width: ${theme.breakpoints.xl}px)`]: {
      flexWrap: 'nowrap',
    },
    [`@media (max-width: ${theme.breakpoints.xs}px)`]: {
      flexWrap: 'wrap',
    },
  }}
>
 ```
 
 I'm curious to know why the less terse option was not included as a design goal from the outset. Thank you for your time :)

**Comment by rtivital** [comment]
We provide style props for all components https://mantine.dev/styles/style-props/
And flex component – https://mantine.dev/core/flex/

**Reply by nk1tz** [reply]
Thanks for the response! I hadn't found this part of the documentation.
Cheers

**Comment by ralrom** [comment]
Hello, I'm new to Mantine and I had the same reflection as @nk1tz

I think the main reason is that the docs have a page called "[Responsive Styles](https://mantine.dev/styles/responsive/)" which doesn't mention this feature. It's only on the next page in the docs menu that this feature is presented. However, if I search "responsive", **Responsive Styles** is the first page that appears in the search results. Maybe it should mention a link to the "Style Props" page?

Specifically, this section made me think the responsive API had no other option:
https://mantine.dev/styles/responsive/#changing-component-size-based-on-media-query
But I understand now that this section refers only to the "size" prop, maybe there should be a mention that other props can be set responsively here?

Just a suggestion based on the confusion I had during my first impressions.
